modular_framework_core/src/modular_framework_core/states/plan.py

```diff
@@ -0,0 +1,203 @@
+
+        elif self.starting_state_type == "pose":
+            starting_robot_state = self.joint_state_to_robot_state(userdata.arm_commander.get_ik(starting_message))
+        else:
+            # In that particular case, the starting robot state is the current robot state during execution
+            starting_robot_state = None
+
+        # Depending on the target state type and the value of the attribute storing the client of the proper service,
+        # determines the target message (either to be retrieved from the manager or from the userdata)
+        if self.get_target_state_message is None and "grasp" in self.target_state_type:
+            target_message = userdata.selected_grasp
+        elif "grasp" in self.target_state_type:
+            target_message = self.get_target_state_message(self.target_state_name).grasp_message
+        elif self.get_target_state_message is None and self.target_state_type == "joint_state":
+            target_message = userdata.selected_joint_state
+        elif self.target_state_type == "joint_state":
+            target_message = self.get_target_state_message(self.target_state_name).joint_state
+        elif self.get_target_state_message is None and self.target_state_type == "pose":
+            target_message = userdata.selected_pose
+        elif self.target_state_type == "pose":
+            target_message = self.get_target_state_message(self.target_state_name).pose_stamped
+        else:
+            rospy.logerr("Impossible to find the required message defining the target state!")
+            return self.outcomes[-1]
+
+        # Compute the plan
+        if self.target_state_type == "pregrasp":
+            plan = userdata.arm_commander.plan_to_pose_target(target_message.pregrasp.pose,
+                                                              custom_start_state=starting_robot_state)
+        elif self.target_state_type == "grasp":
+            plan = userdata.arm_commander.plan_to_pose_target(target_message.grasp.pose,
+                                                              custom_start_state=starting_robot_state)
+        elif self.target_state_type == "postgrasp":
+            plan = userdata.arm_commander.plan_to_pose_target(target_message.postgrasp.pose,
+                                                              custom_start_state=starting_robot_state)
+        elif self.target_state_type == "joint_state":
+            plan = userdata.arm_commander.plan_to_joint_value_target(target_message,
+                                                                     custom_start_state=starting_robot_state)
+        elif self.target_state_type == "pose":
+            plan = userdata.arm_commander.plan_to_joint_value_target(target_message,
+                                                                     custom_start_state=starting_robot_state)
+
+        # This condition checks whether a plan has been found
```

Comment:
things like this really should be wrapped in the correctly names method instead of just having a comment added

Reply:
You mean just for this check or for the other blocks as well?